### PR TITLE
feat: add migration UI and worker [AR-2514]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2514" title="AR-2514" target="_blank">EpicAR-2514</a>  Preserve history from old Android
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

To perform the migration from the old scala app, we need to show the migration screen with progress after starting the app.

### Solutions

Implemented migration screen along with the migration worker, which ensures that even if the user closes the app, the migration process won't be disturbed.

#### How to Test

Right now it's not available as we need to implement the logic of the migration.

### Attachments (Optional)

The migration process is just a delay for the video purposes to show how it looks like.

https://user-images.githubusercontent.com/30429749/193543238-0769dec9-c475-41c7-b7ba-3b909f37e293.mp4

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1058/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1058/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4582319)(4582319)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4582319)(4582319)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/migration/MigrationViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    fun init() {
+        viewModelScope.launch(dispatchers.io()) {
+            enqueueMigrationAndListenForStateChanges()
+        }
+    }
+
+    private suspend fun enqueueMigrationAndListenForStateChanges() {
+        MigrationWorker.enqueue(workManager).collect { state ->
+            if (state == WorkInfo.State.SUCCEEDED)
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/migration/MigrationViewModel.kt]: minor thing, but to me it reads abit weird like you are enqueue a workManager on a MigrationWorker, maybe isntead we can achieve something like : 

```
workManager.enqueueMigrationWorker()
```
 or something like that instead ?

**Review by trOnk12 — APPROVED** [review]
minor thing about the syntax

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1058/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4583914)(4583914)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4583914)(4583914)-dev-release.apk)

**Pull request merged**